sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3898,11 +3898,26 @@ class Dataset[T] private[sql](
    */
   lazy val rdd: RDD[T] = {
     val objectType = exprEnc.deserializer.dataType
+    populateLocalPropertiesInSparkContext(rddQueryExecution.sparkSession)
```

Comment:
This only works if you perform an action RDD immediately afterwards. Isn't this this why we have SQLExecutionRDD? Is so, why is this not wokring?

Reply:
This particular scenario fails only if action is performed after creating an rdd. And in case of SQLExecutionRDD also localProperties are not getting populated.